### PR TITLE
intro_screen/implemented intro screen

### PR DESCRIPTION
**Pull request by aSvetenco**
added
1. AuthActivity that hosts:
IntroFragment - implemented
SignInFragment - just empty fragment
SignUpFragment- just empty fragment

### CODE REVIEW

File: MeetUp/app/src/main/java/com/endava/meetup/screens/auth/modules/intro/IntroFragment.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    public interface IntroCallback {
+        void showSignUp();
+        void showSignIn();
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        try {
+            introCallback = (AuthActivity) context;
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/auth/modules/intro/IntroFragment.java]: what is happening here?

File: MeetUp/app/src/main/java/com/endava/meetup/screens/splash/SplashActivity.java [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
 import com.endava.meetup.R;
+import com.endava.meetup.screens.auth.AuthActivity;
 
 public class SplashActivity extends AppCompatActivity {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_splash);
+        startActivity(new Intent(this, AuthActivity.class));
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/splash/SplashActivity.java]: shouldn't this be called after 4 seconds?

File: MeetUp/app/src/main/java/com/endava/meetup/utils/base/BaseActivity.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+/**
+ * Created by asvetenco on 3/14/2018.
+ */
+
+public abstract class BaseActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        init();
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/utils/base/BaseActivity.java]: why do you need init()? can you do the same login in onCreate in each activity?

File: MeetUp/app/src/main/java/com/endava/meetup/utils/base/BasePresenter.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package com.endava.meetup.utils.base;
+
+/**
+ * Created by asvetenco on 3/14/2018.
+ */
+
+public abstract class BasePresenter<T> {
+
+    T view;
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/utils/base/BasePresenter.java]: why do you need this?

File: MeetUp/app/src/main/java/com/endava/meetup/screens/auth/modules/intro/IntroFragment.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+import com.endava.meetup.screens.auth.AuthActivity;
+
+public class IntroFragment extends Fragment implements IntroFragmentView{
+
+    private IntroCallback introCallback;
+    private IntroFragmentPresenter presenter;
+    private Button signIn;
+    private Button signUp;
+
+    public interface IntroCallback {
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/auth/modules/intro/IntroFragment.java]: Though this is more precise, please call getFragmentManager from BaseFragment, and replace the existing fragment

File: MeetUp/app/src/main/java/com/endava/meetup/screens/splash/SplashActivity.java (hunk 2 above)
Comment (aSvetenco) [reply MeetUp/app/src/main/java/com/endava/meetup/screens/splash/SplashActivity.java]: Pavel will do this...

File: MeetUp/app/src/main/java/com/endava/meetup/screens/auth/modules/intro/IntroFragment.java (hunk 1 above)
Comment (aSvetenco) [reply MeetUp/app/src/main/java/com/endava/meetup/screens/auth/modules/intro/IntroFragment.java]: callback initialization?

**Pull request merged**